obspy/io/rg16/core.py

```diff
@@ -1,233 +1,845 @@
-    ('base_scan', 22, 1, '>i1'),  # https://imgur.com/a/4aneG
-]
-
-# channel set header block
-channel_header_block = [
-    ('chan_num', 1, 1, 'bcd'),
-    ('num_channels', 8, 2, 'bcd'),
-    ('ru_channel_number', 30, 1, '>i1'),
-]
-
-# combines extended header 1, 2, and 3
-extended_header_block = [
-    ('num_records', 16 + 32, 4, '>i4'),
-    ('num_files', 20 + 32, 4, '>i4'),
-    ('collection_method', 32 + 15, 1, '>i1'),
-    ('line_number', 64, 4, '>i4'),
-    ('receiver_point', 68, 4, '>i4'),
-    ('point_index', 69, 1, '>i1'),
-]
-
-# combines trace header blocks 0 (20 byte) and 1 to 10 (32 byte)
-trace_header_block = [
-    ('trace_number', 4, 2, 'bcd'),
-    ('num_ext_blocks', 9, 1, '>i1'),
-    ('line_number', 20 + 0, 3, '>i3'),
-    ('point', 20 + 3, 3, '>i3'),
-    ('index', 20 + 6, 1, '>i1'),
-    ('samples', 20 + 7, 3, '>i3'),
-    ('channel_code', 20 + 20, 1, '>i1'),  # https://imgur.com/a/4aneG
-    ('trace_count', 20 + 21, 4, '>i4'),
-    ('time', 20 + 2 * 32, 8, '>i8'),
-]
-
-# since UTCDateTime cannot be compared to np.inf in py27 get a large timestamp
-# after which I will be dead (somebody else's problem)
-BIG_TS = UTCDateTime('3000-01-01').timestamp
-
-# map sampling rate to band code according to seed standard
-BAND_MAP = {2000: 'G', 1000: 'G', 500: 'D', 250: 'D'}
-
-# geophone instrument code
-INSTRUMENT_CODE = 'P'
-
-# mapping for "standard_orientation"
-STANDARD_COMPONENT_MAP = {'2': 'Z', '3': 'N', '4': 'E'}
+from obspy.io.rg16.util import _read, _open_file
 
 
 @_open_file
 def _read_rg16(filename, headonly=False, starttime=None, endtime=None,
-               merge=False, contacts_north=False, **kwargs):
+               merge=False, contacts_north=False, details=False):
     """
     Read Fairfield Nodal's Receiver Gather File Format version 1.6-1.
-
-    :param filename: A path to the file or a buffer of an opened file.
+    :param filename: path to the rg16 file or a file object.
     :type filename: str, buffer
-    :param headonly: If True don't read data, only header information.
-    :type headonly: bool
-    :param starttime: If not None dont read traces that end before starttime.
+    :param headonly: If True don't read data, only main informations
+     contained in the headers of the trace block are read.
+    :type headonly: optional, bool
+    :param starttime: If not None dont read traces that start before starttime.
     :type starttime: optional, obspy.UTCDateTime
-    :param endtime: If None None dont read traces that start after endtime.
+    :param endtime: If not None dont read traces that start after endtime.
     :type endtime: optional, obspy.UTCDateTime
-    :param merge:
-        If True merge contiguous data blocks as they are found. For
-        continuous data files having 100,000+ traces this will create
-        more manageable streams.
+    :param merge: If True merge contiguous data blocks as they are found. For
+     continuous data files having 100,000+ traces this will create
```

Comment:
multi-line parameter text should be indented an additional level after the first line. See the old code for an example.